src/simulation/navigation/pinholeCamera/pinholeCamera.cpp

```diff
@@ -0,0 +1,328 @@
+
+ */
+
+#include "pinholeCamera.h"
+#include "architecture/utilities/avsEigenSupport.h"
+#include "architecture/utilities/linearAlgebra.h"
+#include "architecture/utilities/rigidBodyKinematics.h"
+#include <iostream>
+#include <math.h>
+#include "architecture/utilities/linearAlgebra.h"
+
+
+/*! @brief Creates an instance of the PinholeCamera class with -90º of Sun's mask angle.
+ @return void
+ */
+PinholeCamera::PinholeCamera()
+{
+    /* Set no lighting conditions by default  */
+    this->maskSun = -MPI_2;
+}
+
+/*! Empty destructor method.
+ @return void
+ */
+PinholeCamera::~PinholeCamera()
+{
+    for (long unsigned int i = 0; i < this->accessOutMsgs.size(); i++){
+        delete this->accessOutMsgs.at(i);
+    }
+    return;
+}
+
+
+/*! Resets the module.*/
+void PinholeCamera::Reset(uint64_t CurrentSimNanos)
+{
+    /* Get number of landmarks */
+    this->n = this->r_LP_P.size();
+    
+    /* Check that required input messages are connected */
+    if (!this->scStateInMsg.isLinked()) {
+        bskLogger.bskLog(BSK_ERROR, "PinholeCamera.scStateInMsg was not linked.");
+    }
+    if (!this->ephemerisInMsg.isLinked()) {
+        bskLogger.bskLog(BSK_ERROR, "PinholeCamera.ephemerisInMsg was not linked.");
+    }
+    
+    /* Compute field of view */
+    this->FOVx = 2 * atan2(this->nxPixel*this->wPixel, 2*this->f);
+    this->FOVy = 2 * atan2(this->nyPixel*this->wPixel, 2*this->f);
+}
+
+/*! Method to add landmarks
+ * @param pos: landmark position in planet-rotating frame
+ * @param normal: landmark surface normal in planet-rotating frame
+ */
+void PinholeCamera::addLandmark(Eigen::Vector3d& pos, Eigen::Vector3d& normal){
+    /* Add the landmark position and surface normal */
+    this->r_LP_P.push_back(pos);
+    this->nL_P.push_back(normal);
+
+    /* Create buffer output messages */
+    Message<AccessMsgPayload> *msg;
+    msg = new Message<AccessMsgPayload>;
+    this->accessOutMsgs.push_back(msg);
+
+    /* Expand the access buffer vectors */
+    AccessMsgPayload accMsg;
+    this->accessMsgBuffer.push_back(accMsg);
+}
+
+/*! Loop through landmarks
+*/
+void PinholeCamera::loopLandmarks()
+{
+    /* Define variables */
+    Eigen::Vector3d rLmk, nLmk;
+    Eigen::Vector2i pLmk;
+    bool flagFOV, flagLighting;
+    
+    /* Preallocate visibility status and landmark pixels */
+    this->isvisibleLmk.setZero(this->n);
+    this->pixelLmk.setZero(this->n, 2);
+    
+    /* Initialize visible landmarks */
+    this->nvisibleLmk = 0;
+    
+    /* Loop through landmarks */
+    for (int i = 0; i < this->n; i++){
+        /* Extract ith landmark position and its normal */
+        rLmk = this->r_LP_P[i];
+        nLmk = this->nL_P[i];
+        
+        /* Compute pixel and check access conditions */
+        pLmk = this->computePixel(rLmk);
+        flagFOV = this->checkFOV(pLmk, rLmk, nLmk);
+        flagLighting = this->checkLighting(nLmk);
+        
+        /* Add pixel */
+        if (flagFOV and flagLighting){
```

Comment:
If I'm not mistaken we don't enable the the `/permissive` or `/Za` flags on windows and are unlikely to at this moment. So you could fix you windows CI build by not using the `and` keyword, but rather `&&`.

Reply:
Thank you for signalling the issue. Will try to see if that fixes it as there are multiple complaints in the windows build